src/Tinker.jl

```diff
@@ -121,360 +131,77 @@ function drawrecthandle(ctx, rh::RectHandle, d, color1, width)
-    end
-    return p2
+# Moves polygon to a given location
+function move_polygon_to(p::AbstractArray, pt::XY)
+    # find diff b/t start & pt; add diff to all in p
+    diff = XY(pt.x-p[1].x,pt.y-p[1].y)
+    map(n -> XY(n.x+diff.x,n.y+diff.y),p)
+end
+
+function Point(p::XY)
+    Point(p.x,p.y)
 end
 
-# Mouse actions for rectangular selection creation and modification
-function init_rect_select(ctx::ImageContext)
+function init_move_polygon(ctx::ImageContext)
     c = ctx.canvas
-    rect = ctx.shape
-    
-    recthandle = map(rect) do r
-         RectHandle(r)
-    end
+    pts = ctx.points
     enabled = Signal(true)
-    dragging = Signal(false) # true if mouse was pressed down before it was moved
-    modifying = Signal(false) # true if there is an existing rectangle &
-    # user clicked somewhere on it
-    p1 = Signal(XY{UserUnit}(-1.0,-1.0)) # ints or floats? -> make ints
-    p2 = Signal(XY{UserUnit}(-1.0,-1.0))
-    modhandle = Signal(Handle()) # handle that was clicked
-    locmod = Signal(false) # true if modifying rectangle location
-    diff = Signal(XY(0.0,0.0)) # difference between btn.position of sigstart &
-    # top left corner of rectangle - used for moving rect
-
-    dummybtn = MouseButton{UserUnit}()
-    sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
-        push!(dragging, true)
-        # If there is already a rectangle in the environment, begin to modify
-        if !isempty(value(recthandle))
-            push!(modifying,true)
-            # Identify if click is inside handle
-            current = Handle()
-            d = 8*(IntervalSets.width(value(ctx.zr).currentview.x)/IntervalSets.width(value(ctx.zr).fullview.x)) # physical dimension of handle
-            for n in 1:length(value(recthandle).h)
-                if is_clicked(btn.position, value(recthandle).h[n], d)
-                    current = value(recthandle).h[n]
-                    push!(modhandle, current)
-                    break
-                end
-            end
-            # If the click was on a handle:
-            if !isempty(current)
-                push!(p1, get_p1(current, value(recthandle)))
-                push!(p2, get_p2(current, value(recthandle), btn))
-            elseif (value(rect).x<btn.position.x<value(rect).x+value(rect).w) &&
-                (value(rect).y<btn.position.y<value(rect).y+value(rect).h)
-                push!(locmod,true)
-                # the difference between click and rectangle corner
-                push!(diff, XY(Float64(btn.position.x) - value(rect).x,
-                               Float64(btn.position.y) - value(rect).y))
-            else
-                # Destroy rectangle and allow for a new one to be created
-                push!(modifying,false)
-                push!(rect, Rectangle())
-                push!(p1, btn.position)
-                push!(p2, btn.position)
+    dragging = Signal(false)
+    diff = Signal(XY(-1.0,-1.0))
```

Comment:
Maybe use `NaN` instead of `-1.0` since `-1.0` seems like a perfectly valid value for `diff`.